pkg/terraform/state/handler.go

```diff
@@ -83,6 +86,20 @@ func (h *handler) OnChange(key string, obj *v1.State) (*v1.State, error) {
 		return h.states.Update(obj)
 	}
 
+	now := metaV1.Now()
+
+	if scheduleRefresh(obj, &now) && v1.StateConfitionScheduled.IsFalse(obj) {
+		duration, err := time.ParseDuration(obj.Annotations[RefreshAnnotationKey])
+		if err != nil {
+			logrus.Error(err)
```

Comment:
If duration was invalid,  what value is scheduled  `now.Add(0)`?;   consider returning function here to better readability and consecutive protection

```suggestion
logrus.Error(err)
return ....
```

Reply:
@kuritka done